simulations/src/node/carnot/researsh_overlay.rs

```diff
@@ -0,0 +1,98 @@
+use consensus_engine::{overlay::LeaderSelection, Committee, NodeId, TimeoutQc};
+
+use crate::node::CommitteeExt;
+
+#[derive(Debug, Clone, Copy, serde::Serialize, serde::Deserialize)]
+pub struct ResearchOverlaySettings<L: LeaderSelection> {
+    num_nodes: usize,
+    leader: L,
+    leader_threshold: usize,
+}
+
+#[derive(Debug, Clone)]
+pub struct ResearchOverlay<L: LeaderSelection> {
```

Comment:
How is this different from the `Flatoverlay`. IIRC what we wanted to have is a linked list of committees that simulates a single branch of a tree overlay.

Reply:
Very similar. The different part is how to construct the overlay. In FlatOverlay, we pass a set of nodes, whereas, in ResearchOverlay, the nodes are generated according to the total levels.

Reply:
I do not get it. That can be totally decoupled to use the same system we already have 🤔